sphinx/builders/html/__init__.py

```diff
@@ -1035,8 +1035,21 @@ def hasdoc(name: str) -> bool:
             templatename = newtmpl
 
         # sort JS/CSS before rendering HTML
-        ctx['script_files'].sort(key=lambda js: js.priority)
-        ctx['css_files'].sort(key=lambda js: js.priority)
+        try:
+            ctx['script_files'] = sorted(list(ctx['script_files']), key=lambda js: js.priority)
+        except AttributeError:
```

Comment:
I don't think that with this change an `AttributeError` will ever happen... Well, it may happen on `js.priority` but that's a different problem.

Reply:
Yes. That is #8885.